DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsControllerTests.cs

```diff
@@ -172,13 +184,162 @@ public void AddRegistrationPromptSelectPrompt_loads_existing_temp_data()
         {
             var expectedTempData = new AddRegistrationPromptData();
             registrationPromptsControllerWithMockHttpContext.TempData.Set(expectedTempData);
-            
+
             // When
             var result = registrationPromptsControllerWithMockHttpContext.AddRegistrationPromptSelectPrompt();
 
             // Then
-            registrationPromptsControllerWithMockHttpContext.TempData.Peek<AddRegistrationPromptData>().Should().BeEquivalentTo(expectedTempData);
+            registrationPromptsControllerWithMockHttpContext.TempData.Peek<AddRegistrationPromptData>().Should()
+                .BeEquivalentTo(expectedTempData);
             result.Should().BeViewResult().WithDefaultViewName();
         }
+
+        [Test]
+        public void AddRegistrationPromptSelectPrompt_post_updates_temp_data_and_redirects()
+        {
+            var expectedPromptModel = new AddRegistrationPromptSelectPromptViewModel
+            {
+                CustomPromptId = 1,
+                Mandatory = true
+            };
+            var initialTempData = new AddRegistrationPromptData();
+            registrationPromptsController.TempData.Set(initialTempData);
+
+            // When
+            var result = registrationPromptsController.AddRegistrationPromptSelectPrompt(expectedPromptModel);
+
+            // Then
+            registrationPromptsController.TempData.Peek<AddRegistrationPromptData>().Should().NotBeNull();
+            registrationPromptsController.TempData.Peek<AddRegistrationPromptData>()!.SelectPromptViewModel.Should()
+                .BeEquivalentTo(expectedPromptModel);
+            result.Should().BeRedirectToActionResult().WithActionName("AddRegistrationPromptConfigureAnswers");
+        }
+
+        [Test]
+        public void AddRegistrationPromptConfigureAnswers_next_updates_temp_data()
+        {
+            // Given
+            var expectedPromptModel = new AddRegistrationPromptSelectPromptViewModel
+            {
+                CustomPromptId = 1,
+                Mandatory = true
+            };

```

Comment:
Could this benefit from a constructor for more concise declaration? (+ elsewhere to make it easier to read)

Reply:
I think you missed L168 👁️